Issue #242: Address lower API CSS issues with uploading image containers

aforcier opened the pull request:
Fixes #242, addressing the following bugs that appear on `API < 19` only:
- Fixes the issue where the uploading image container would sometimes be very small: https://cloudup.com/c2HhsInzXsW
- Because the `API < 19` `WebView` doesn't support the progress tag, there is no indication that the image is uploading. For those API levels, an 'Uploading...' overlay is shown instead:

![242-upload-overlay-text-old-api](https://cloud.githubusercontent.com/assets/9613966/10137628/c15c06e4-65fa-11e5-9bbd-4f4b6e0c9ed1.png)

Some explanation about 9e1863f: Fixing the initial issue with the small containers required using `display: block` instead of `display: inline-block`. But this makes the container fill the screen width rather than match the image size, so for images less wide than the screen the overlays are skewed:

![242-uploading-image-block-bug-api18](https://cloud.githubusercontent.com/assets/9613966/11458278/e2a296f4-968a-11e5-855a-4fab76db6d99.png)

I found that once the image has loaded (on progress updates after the initial one which merely initializes the progress bar), we can safely drop back to `inline-block` on those API levels. This keeps the small container fix working, but also properly centers the overlays on the image for smaller images:

![242-uploading-image-inline-block-api18](https://cloud.githubusercontent.com/assets/9613966/11458291/0e41b236-968b-11e5-9387-dbaa15b296a5.png)

It's not an ideal fix as we may occasionally see a brief alignment issue until the first progress update comes in, but I haven't found a better way to fix both things at once so far.

cc @bummytime @thianhlu

aforcier committed:
Updated ZSSEditor to get the current Android API level on init

aforcier committed:
On pre-KitKat Android, show an 'Uploading...' overlay instead of a progress bar for uploading images

aforcier committed:
Use block instead of inline-block for the image container on pre=KitKat Android

- This fixes a display issue where the image container would sometimes be very small during upload (pre-KitKat only)

aforcier committed:
Hide the compatibility image upload overlay if an image upload fails and re-display it on resume

aforcier committed:
Revert to non-compatibility image container styling once image has begun uploading

- This keeps the uploading and failed overlays centered on the image

bummytime commented on libs/editor-common/assets/ZSSRichTextEditor.js:
```diff
@@ -932,6 +946,12 @@ ZSSEditor.setProgressOnImage = function(imageNodeIdentifier, progress) {
+    // Revert to non-compatibility image container once image upload has begun. This centers the overlays on the image
+    // (instead of the screen), while still circumventing the small container bug the compat class was added to fix
+    if (progress > 0) {
+        this.getImageContainerNodeWithIdentifier(imageNodeIdentifier).removeClass("compat");
```
:sparkles: :ghost:

bummytime commented:
Nice fix @aforcier. It runs well on a variety of low-API devices and code looks fine. The only comment I have is regarding the text on a light image:

<img width="150" alt="scaled screen shot 2015-11-30 at 3 06 23 pm" src="https://cloud.githubusercontent.com/assets/154014/11484573/5acf124e-9774-11e5-9521-4b1abf9163bb.png">

Can be difficult to see. Any thoughts?

aforcier commented:
> The only comment I have is regarding the text on a light image...can be difficult to see. Any thoughts?

I think we can use a semi-transparent dark overlay over the image while uploading again, to increase the contrast for the `Uploading...` message. This is what we did for the [edit overlay](https://github.com/wordpress-mobile/WordPress-Editor-Android/pull/244#issuecomment-139531939).

![13-edit-image-api18-dark-overlay](https://cloud.githubusercontent.com/assets/9613966/9814622/16b78908-5898-11e5-9b6b-8b961be9e066.png)

I have this mostly working, there are some CSS quirks to resolve first. I'll update this branch when I get that working.

aforcier committed:
Added a dark semi-transparent overlay for API<19 for uploading and failed images

- This gives better visibility to the 'Uploading...' and 'Retry' text, especially for lightly-colored images

aforcier committed:
Fixed a display issue on API<19 where a dark blank space would appear below an uploading or failed image

aforcier committed:
Fixed issue where compatibility upload overlays would sometimes consume a tap without triggering a tap callback for the associated image

- This caused 'Retry' taps to have no effect on API<19 devices

aforcier commented:
Added a dark semi-transparent overlay on `API<19` for uploading and failed images.

Ready for another look @bummytime!

aforcier committed:
Merge branch 'develop' into issue/242-image-container-lower-api

Conflicts:
	libs/editor-common/assets/editor-android.css

aforcier committed:
Merge branch 'develop' into issue/242-image-container-lower-api

Conflicts:
	WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java

aforcier committed:
Used a String constant for naming the native state JS interface

aforcier committed:
Fixed some EditorFragment lint issues

tonyr59h commented:
Looks good, @aforcier! Working as expected above and below the API 19 line.

:shipit:
